service/controller/v21/adapter/guest_outputs.go

```diff
@@ -23,11 +23,13 @@ func (a *GuestOutputsAdapter) Adapt(config Config) error {
 
 	a.Worker.ASG.Key = key.WorkerASGKey
 	a.Worker.ASG.Ref = key.WorkerASGRef
+	a.Worker.CloudConfig.Version = config.StackState.WorkerCloudConfigVersion
 	a.Worker.Count = config.StackState.WorkerCount
 	a.Worker.DockerVolumeSizeGB = strconv.Itoa(config.StackState.WorkerDockerVolumeSizeGB)
 	a.Worker.ImageID = config.StackState.WorkerImageID
 	a.Worker.InstanceType = config.StackState.WorkerInstanceType
-	a.Worker.CloudConfig.Version = config.StackState.WorkerCloudConfigVersion
+	a.Worker.Max = config.StackState.WorkerMax
+	a.Worker.Min = config.StackState.WorkerMin
```

Comment:
What is the reason to track this in the outputs? During stack state transitioning the output values might not be accessible IIRC. Getting the values from the ASG when we need it would be more reliable as the occasionally unavailable redirect here, wouldn't it?

Reply:
Stackstate transitioning is the mechanism involved also in scaling vs. update decisions, but you have a good point. I'll do some investigation and see if we could get rid of this. Getting this right was slightly difficult and I now have even a bug in there (due to nil value for `WorkerCount`) so I'll take a look at this.

Reply:
Ok. I'm working on this. I'm pretty sure I can get rid of tracking this via CloudFormation stack output, but I keep the state in `StackState` still, it's just read directly from ASG instead of stack output. This way I can also read the present `Desired` property at the right time and later just save it in Status field in `ApplyUpdate()` -> no need for additional resource then. I'm still in the middle of finalizing this, but so far it looks good. Thanks for your comment here! :heart:

Reply:
Staying in the same resource makes sense when the lifecycle of information and behaviour is actually aligned.